sound/soc/sof/intel/cnl.c

```diff
@@ -233,14 +233,44 @@ void cnl_ipc_dump(struct snd_sof_dev *sdev)
 		hipcida, hipctdr, hipcctl);
 }
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_PROBES)
+DEFINE_IDA(sof_probes_client_ida);
+
+static int hda_register_probes_client(struct snd_sof_dev *sdev)
+{
+	return sof_client_dev_register(sdev, "probes", &sof_probes_client_ida);
+}
+
+static void hda_unregister_probes_client(struct snd_sof_dev *sdev)
+{
+	sof_client_dev_unregister(sdev, "probes", 0);
```

Comment:
do we actually need an IDA for probes? You could simplify since we only support one!